src/code_80850_len_3060.c

```diff
@@ -268,7 +268,108 @@ void enforce_hpfp_limits(void) {
     }
 }
 
-INCLUDE_ASM(s32, "code_80850_len_3060", initialize_status_menu);
+void initialize_status_menu() {
+    UiStatus* uiStatus = &gUIStatus;
+    PlayerData* playerData = &gPlayerData;
```

Comment:
You should be able to use UI_STATUS and PLAYER_DATA macros here. If the UI_STATUS one doesn't exist, feel free to create it in macros.h like the player data one